### ISSUE TITLE
No logs

### ISSUE THREAD

**Issue by verisgit** [hjdhjd/homebridge-unifi-protect#839]
**Describe The Problem:**
Keen to try out this plugin, and it seems very extensive.

I have installed the plugin per the instructions and configured it to point at my controller. I've tried it with and without admin enabled.

I can't see anything in the logs to indicate what isn't working; however, I have nothing showing up in Apple Home and I can't see any log entries to indicate that the plugin is logging into the controller. The usernames/passwords have been verified and I've logged into the controller manually with the local access to check it has access to the streams.

Not sure what the issue is given I can't see any logs, so not sure what to try or where to start. Is there a verbose logging mode I can enable?

**To Reproduce:**
N/A - no output

**Logs:**

```
[11/20/2022, 10:22:23 PM] Loaded plugin: homebridge-unifi-protect@5.5.4 
[11/20/2022, 10:22:23 PM] Registering platform 'homebridge-unifi-protect.UniFi Protect'
```

**Homebridge Configuration:**

```json
{
            "controllers": [
                {
                    "address": "MYIP",
                    "password": "MYPASS",
                    "username": "MYUSERNAME"
                }
            ],
            "platform": "UniFi Protect"
        },
```

**Screenshots:**
N/A

**Environment:**

* **Homebridge Version:** 1.5.1
* **Node Version:** 13.10.1
* **Homebridge-UniFi-Protect Plugin Version:** 5.5.4
* **UniFi Protect Version:** 2.2.6
* **UniFi Protect Controller Type:** UDMPRO
* **UniFi Protect Controller Firmware Version:** 1.12.33
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:**16.0.3
* **Operating System and OS Version:** FreeBSD (TrueNAS Homebridge Jail) - with FFMPEG installed and working (used with Homebridge Camera FFmpeg)

**FFMPEG Info**
```
ffmpeg version 4.2.3 Copyright (c) 2000-2020 the FFmpeg developers
  built with FreeBSD clang version 8.0.0 (tags/RELEASE_800/final 356365) (based on LLVM 8.0.0)
  configuration: --prefix=/usr/local --mandir=/usr/local/man --datadir=/usr/local/share/ffmpeg --pkgconfigdir=/usr/local/libdata/pkgconfig --enable-shared --enable-pic --enable-gpl --enable-avresample --cc=cc --disable-alsa --disable-libopencore-amrnb --disable-libopencore-amrwb --enable-libaom --disable-libaribb24 --enable-libass --disable-libbs2b --disable-libcaca --disable-libcdio --disable-libcelt --disable-libcodec2 --enable-libdav1d --disable-libdavs2 --disable-libdc1394 --disable-debug --enable-htmlpages --disable-libdrm --disable-libfdk-aac --disable-libflite --enable-fontconfig --enable-libfreetype --enable-frei0r --disable-libfribidi --disable-gcrypt --disable-libgme --enable-gmp --enable-gnutls --enable-version3 --disable-libgsm --enable-iconv --disable-libilbc --disable-libjack --disable-libklvanc --disable-libkvazaar --disable-ladspa --disable-libmp3lame --disable-liblensfun --disable-libbluray --disable-librsvg --disable-librtmp --disable-libxml2 --disable-lto --disable-lv2 --disable-mbedtls --disable-libmfx --enable-mmx --disable-libmodplug --disable-libmysofa --disable-nonfree --disable-openal --disable-opencl --enable-libopencv --disable-opengl --disable-libopenh264 --disable-libopenjpeg --disable-libopenmpt --disable-openssl --enable-optimizations --enable-libopus --disable-pocketsphinx --disable-libpulse --disable-librav1e --enable-runtime-cpudetect --disable-librubberband --disable-sdl2 --disable-libsmbclient --disable-libsnappy --disable-sndio --disable-libsoxr --disable-libspeex --disable-libsrt --enable-sse --disable-libssh --disable-libtensorflow --disable-libtesseract --enable-libtheora --disable-libtwolame --enable-libv4l2 --enable-vaapi --disable-vapoursynth --enable-vdpau --disable-libvidstab --disable-libvmaf --enable-libvorbis --disable-libvo-amrwbenc --enable-libvpx --disable-libwavpack --disable-libwebp --enable-libx264 --enable-libx265 --disable-libxavs2 --enable-libxcb --enable-libxvid --disable-outdev=xv --disable-libzimg --disable-libzmq --disable-libzvbi
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
Hyper fast Audio and Video encoder
```

**Comment by hjdhjd** [comment]
Everything looks correct in the above - I’ve actually got a nearly identical configuration.

Obvious things that you may have already tried / checked:

1. Are you sure you’ve got Homebridge running correctly, including getting it added to your HomeKit home?
2. Are you certain you’ve properly configured the IP address of the Protect controller, as well as username and password?

I’m assuming the answers to 1 is yes, given you’ve got homebridge-camera-ffmpeg working. 😄 
2…probably the single biggest misconfiguration issue I tend to see.

This plugin produces a prodigious amount of logging by default. You don’t want to make it more verbose, as a general rule.

Best of luck.

**Comment by verisgit** [comment]
1. yes
2. yes

I just installed a brand new FreeBSD jail for testing with a brand new homebridge. Added it to apple home. Setup homebridge-unifi-protect and again I get nothing in the logs, it never connects to the controller, and I get no devices in Homekit.

I'm assuming if it can't connect to the controller I would see something in the logs. Given I'm not seeing anything in the logs and that's the first step - I'm really confused as to what's going on.

How is the plugin connecting to the controller? Is there a unifi API? Can I test the connection manually via curl?

**Comment by verisgit** [comment]
I did some digging in the unifi forums.. seems there's a bug with local accounts and they sometimes simply don't work despite everything in the UI indicating they are fine. So I created a bunch of local accounts, logged into all of them manually and then tried one of the test accounts with admin enabled and it finally started producing logs (hundreds of lines of JSON). Seems to be working fine with that account but not with two of the others even though they are all identically created. Very strange but glad it's now working and it's exceptionally good so many thanks for the hard work!
